msticpy/nbtools/morph_charts.py

```diff
@@ -0,0 +1,156 @@
+        Path.mkdir(Path.cwd().joinpath("morphchart_package"), exist_ok=True)
+        # Write description file
+        morph_file = open(out_path, "w")
+        json.dump(description_dict, morph_file)
+        # Write dataset to query_data csv
+        data_out_path = out_path = Path.cwd().joinpath(
+            *["morphchart_package", "query_data.csv"]
+        )
+        data.to_csv(data_out_path, index=False)
+        # Display Morph Charts in IFrame with instructions
+        print(
+            f"Navigate to {Path.cwd().joinpath('morphchart_package')} and upload the files below"
+        )
+        print("Charts provided by http://morphcharts.com/")
+        return IFrame("http://morphcharts.com/designer.html", "100%", "600px")
+
+    def list_charts(self):
+        """Get a list of avaliable charts."""
+        for key, _ in self.charts.items():
+            print(key)
+
+    def get_chart_details(self, chart_name):
+        """
+        Get description for a chart.
+
+        Parameters
+        ----------
+        chart_name: str:
+            The name of the chart you get description for.
+
+        """
+        try:
+            print(
+                chart_name,
+                ":",
+                "\n",
+                self.charts[chart_name]["Description"],
+                "\n",
+                "Query: ",
+                self.charts[chart_name]["Query"],
+            )
+        except KeyError:
+            raise KeyError(f"Unknown chart {chart_name}")
+
+    def search_charts(self, keyword):
+        """
+        Search for charts that match a keyword.
+
+        Parameters
+        ----------
+        keyword: str:
+            The keyword to search charts for.
+
+        """
+        for key, value in self.charts.items():
+            if keyword.casefold() in [tag.casefold() for tag in value["Tags"]]:
+                print(key, ":", "\n", self.charts[key]["Description"])
+            elif keyword.casefold() in [
+                word.casefold() for word in value["Description"].split()
+            ]:
+                print(key, ":", "\n", self.charts[key]["Description"])
+            else:
+                print("No matching charts found")
+
+
+def _get_charts(path: str = "MorphCharts") -> dict:
+    """
+    Return dictionary of yaml files found in the Morph Charts folder.
+
+    Parameters
+    ----------
+    path : str
+        The source path to search in.
+
+    Returns
+    -------
+    Dict
+        Details of the chart files
+
+    """
+    full_path = Path(__file__).parent.parent.joinpath("data").joinpath(path)
+    file_glob = Path(full_path).glob(f"*.yaml")
+    chart_files = [file_path for file_path in file_glob if file_path.is_file()]
+    chart_details = {}
+    for chart in chart_files:
+        chart_data = open(chart, "r")
+        details = yaml.safe_load(chart_data)
+        chart_details.update(
+            {
+                details["Name"]: {
```

Comment:
Might want to give more feedback about input given that we might not control the file. I guess the dict key failure would would tell people that they didn't have the correct file format. Perhaps just wrap this in a try/except and throw an exception saying that the file is invalid format.